plugins/performance-lab/includes/site-health/far-future-headers/helper.php

```diff
@@ -0,0 +1,183 @@
+<?php
+/**
+ * Helper function to detect if static assets have far-future expiration headers.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Callback for the far-future caching test.
+ *
+ * @since n.e.x.t
+ *
+ * @return array{label: string, status: string, badge: array{label: string, color: string}, description: string, actions: string, test: string} Result.
+ */
+function perflab_ffh_assets_test(): array {
+	$result = array(
+		'label'       => __( 'Your site serves static assets with far-future expiration headers', 'performance-lab' ),
+		'status'      => 'good',
+		'badge'       => array(
+			'label' => __( 'Performance', 'performance-lab' ),
+			'color' => 'blue',
+		),
+		'description' => sprintf(
+			'<p>%s</p>',
+			esc_html__(
+				'Serving static assets with far-future expiration headers improves performance by allowing browsers to cache files for a long time, reducing repeated requests.',
+				'performance-lab'
+			)
+		),
+		'actions'     => '',
+		'test'        => 'is_far_future_headers_enabled',
+	);
+
+	// List of assets to check.
+	$assets = array(
+		plugins_url( 'far-future-headers/assets/test.css', __DIR__ ),
+	);
```

Comment:
>Also, there could be some cases where an author needs for an asset to not have a far-future expiration date. In that case, there could be a filter to exclude it from the test.

https://github.com/WordPress/performance/issues/323#issuecomment-1121307393

Currently we can add assets which we want to test in this array and we could convert it to filter as suggested in the comment. 
But I have a doubt should we add the test files into our plugin for testing different assets ( `CSS`, `JS`, `images`, `fonts` ) like I have added the `test.css`? or do we need to test the enqueued assets here?

Reply:
We can reuse static assets which are located in WordPress core. So you can obtain the URLs for those files located in `wp-includes` (or `wp-admin`).